src/main/java/core/basesyntax/dao/StorageDaoImpl.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.storage.Storage;
+import java.util.Map;
+
+public class StorageDaoImpl implements StorageDao {
+    private Storage storage;
+
+    public StorageDaoImpl() {
+        storage = new Storage();
+    }
+
+    @Override
+    public Map<String, Integer> save(String fruit, int quantity) {
+        storage.getFruitStorage().put(fruit, quantity);
+        return storage.getFruitStorage();
```

Comment:
The `save` method returns the entire storage map, which might not be necessary if the purpose is only to update the storage. Consider returning a more appropriate type or void if the map is not needed after saving.